zoloto/coords.py

```diff
@@ -1,13 +1,84 @@
-from typing import NamedTuple
+from typing import Iterator, NamedTuple, Tuple
 
+from cached_property import cached_property
 from coordinates import spaced_coordinate
+from cv2 import Rodrigues
+from numpy import ndarray
+from pyquaternion import Quaternion
 
 Coordinates = spaced_coordinate("Coordinates", "xy")
 
-Orientation = NamedTuple(
-    "Orientation", [("rot_x", float), ("rot_y", float), ("rot_z", float)]
-)
-
 ThreeDCoordinates = spaced_coordinate("ThreeDCoordinates", "xyz")
 
 Spherical = NamedTuple("Spherical", [("rot_x", float), ("rot_y", float), ("dist", int)])
+
+
+class Orientation:
+    """The orientation of an object in 3-D space."""
+
+    def __init__(self, *rvec) -> None:
+        rotation_matrix, _ = Rodrigues(rvec)
+        self._quaternion = Quaternion(matrix=rotation_matrix)
+
+    @property
+    def rot_x(self) -> float:
+        """Get rot_x aka. roll."""
```

Comment:
I assume all these values are in radians? Would be nice to note that in the docstring

Reply:
3373076